File: iact-check-webapp/src/main/resources/webapp/src/app/shared/dtos/possible-outcome-dto.ts
import { PossibleScoreDTO } from './possible-score-dto';

export interface PossibleOutcomeDTO {
  id: number;
  questionCategoryId: number;
  title: string;
  subtitle: string;
  description: string;
  youtubeUrl: string;
  backgroundColour: string;
  possibleScores: PossibleScoreDTO[];
}

File: iact-check-core/src/main/kotlin/ch/iact/iactcheck/dto/PossibleOutcomeDTO.kt
package ch.iact.iactcheck.dto

data class PossibleOutcomeDTO(
    val id: Long,
    val questionCategoryId: Long,
    val title: String,
    val subtitle: String,
    val description: String,
    val youtubeUrl: String? = null,
    val backgroundColour: String? = null,
    val possibleScores: List<PossibleScoreDTO>
)
